lib/paxful_client/requests/base_request.rb

```diff
@@ -0,0 +1,27 @@
+module PaxfulClient
+  class BaseRequest
+
+    include APIClientBase::Request
+
+    attribute :key,    String
+    attribute :secret, String
+
+    private
+
+    def body
+      nonce = Time.now.to_i
+      apiseal = OpenSSL::HMAC.hexdigest(
+        "SHA256", 
+        secret, 
+        "apikey=#{key}&nonce=#{nonce}",
+      )
+
+      [
+        "apikey=#{key}",
+        "nonce=#{nonce}",
+        "apiseal=#{apiseal}",
+      ].join("&")
+    end
```

Comment:
You can use Addressable::URI to more easily build this. There are methods around params

Reply:
Oh I see you're just moving it, nvm